src/Render.cc

```diff
@@ -57,38 +61,52 @@ namespace drafter {
            attributes = &actionAttributes;
         }
 
+        // Only continue down if we have a render format
         if (!payload.node->body.empty() || attributes->node->empty() || renderFormat == UndefinedRenderFormat) {
             return body;
         }
 
+        // Expand MSON into Refract
+        refract::IElement* element = MSONToRefract(*attributes);
+
+        if (!element) {
+            return body;
+        }
+
+        refract::IElement* expanded = ExpandRefract(element, registry);
+
+        if (!expanded) {
+            return body;
+        }
+
+        // One of this will always execute since we have a catch above for not having render format
         switch (renderFormat) {
             case JSONRenderFormat:
             {
                 refract::RenderJSONVisitor renderer;
+                renderer.visit(*expanded);
 
-                refract::IElement* element = MSONToRefract(*attributes);
-
-                if (!element) {
-                    return body;
-                }
-
-                refract::IElement* expanded = ExpandRefract(element, registry);
+                delete expanded;
 
-                if (!expanded) {
-                    return body;
-                }
+                return std::make_pair(renderer.getString(), NodeInfo<Asset>::NullSourceMap());
+            }
 
-                renderer.visit(*expanded);
+            case JSONSchemaRenderFormat:
+            {
+                refract::JSONSchemaVisitor renderer;
+                std::string result = renderer.getSchema(*expanded);
 
                 delete expanded;
 
-                return std::make_pair(renderer.getString(), NodeInfo<Asset>::NullSourceMap());
+                return std::make_pair(result, NodeInfo<Asset>::NullSourceMap());
             }
 
             default:
                 break;
         }
 
+        // Just to be cautious
```

Comment:
We should never get here so I would say abort here, the same for the default switch case, it should never happen so either abort or throw some runtime exception, we should know if this ever happens and not silently let it go through